### ISSUE TITLE
Build and deployment server

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/chipper#141]
Input: a list of repo dependencies and the SHAs that they should be at, and list of locales to build
Probably should include where to publish (dev? production?)

Step 1: Checkout repos with SHAs (if necessary), npm install (if necessary)
Step 2: Build the specified locales
Step 3: Upload the built files to the desired server(s)
Step 4: Notify said servers (websites) that they were deployed

The request for build/deploy should be authenticated somehow (it's a PhET request), and any errors should stop the
entire process (and hopefully return a status result about the build).

It would be nice for it to be able to build without deploying anything (for continuous integration testing), just to
see if there are any build problems introduced by commits.

Then we could have a lightweight chipper script that could fire off the request to the build server.

Additionally, it would be nice to have the default build/deploy first build and copy the files, but not "deploy" as in
overwrite the currently-live version. It would be nice to run and test the HTML files that have been built before that
step happening (like the translation deployment).

**Comment by aaronsamuel137** [comment]
@jonathanolson, are imaging a queue here so that only one deploy can be happening at a time? It seems like we can't run them in parallel since different shas will need to be checked out for each deploy request.

**Comment by jonathanolson** [comment]
Yes, that is what I was imagining.

**Comment by aaronsamuel137** [comment]
I think version 1 of build server is ready for code review. Assigning to @jonathanolson for review.

**Comment by jonathanolson** [comment]
Review completed. In great shape overall, comment level is perfect, and it has way better error handling than anything we've had for Java/Flash.

Some issues that could affect the build-server the most:
- Separate location for build-server, and possibly separate repo: https://github.com/phetsims/chipper/issues/345
- Alternative deployment support: https://github.com/phetsims/chipper/issues/343
- Assumptions about future chipper behavior/structure: https://github.com/phetsims/chipper/issues/342

The rest are questions or would potentially require only small/localized changes. I'd recommend scanning the above three issues first.

Assigning to @aaronsamuel137 to look into the review issues.

**Comment by aaronsamuel137** [comment]
Review was completed awhile ago. Closing
